assets/js/googlesitekit/datastore/site/developer-plugin.test.js

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import {
+	createTestRegistry,
+	muteFetch,
+	subscribeUntil,
+	unsubscribeFromAll,
+} from 'tests/js/utils';
+import { STORE_NAME } from './constants';
+
+describe( 'core/site developer plugin state', () => {
+	const responseDeveloperPluginState = {
+		active: true,
+		installed: true,
+		activateURL: 'http://something.test',
+		installURL: 'http://something.test',
+		configureURL: 'http://something.test',
+	};
+
+	let registry;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	describe( 'actions', () => {
+		it( 'does not require any params', () => {
+			expect( () => {
+				muteFetch( /^\/google-site-kit\/v1\/core\/site\/data\/developer-plugin/ );
+				registry.dispatch( STORE_NAME ).fetchGetDeveloperPluginState();
+			} ).not.toThrow();
+		} );
+
+		describe( 'receiveGetDeveloperPluginState', () => {
+			it( 'requires the response param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).receiveGetDeveloperPluginState();
+				} ).toThrow( 'response is required.' );
+			} );
+
+			it( 'receives and sets developer plugin state', async () => {
+				await registry.dispatch( STORE_NAME ).receiveGetDeveloperPluginState( responseDeveloperPluginState );
+
+				const { developerPluginState } = registry.stores[ STORE_NAME ].store.getState();
+
+				expect( developerPluginState ).toMatchObject( responseDeveloperPluginState );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getDeveloperPluginState', () => {
+			it( 'uses a resolver to make a network request', async () => {
+				fetchMock.getOnce(
+					/^\/google-site-kit\/v1\/core\/site\/data\/developer-plugin/,
+					{ body: responseDeveloperPluginState, status: 200 }
+				);
+
+				const initialDeveloperPluginState = registry.select( STORE_NAME ).getDeveloperPluginState();
+				// The developer plugin state will be its initial value while the developer plugin state is fetched.
+				expect( initialDeveloperPluginState ).toEqual( undefined );
+				await subscribeUntil( registry,
+					() => (
+						registry.select( STORE_NAME ).getDeveloperPluginState() !== undefined
+					),
+				);
+
+				const developerPluginState = registry.select( STORE_NAME ).getDeveloperPluginState();
+
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+				expect( developerPluginState ).toEqual( responseDeveloperPluginState );
+
+				const developerPluginStateSelect = registry.select( STORE_NAME ).getDeveloperPluginState();
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+
+				expect( developerPluginStateSelect ).toEqual( developerPluginState );
```

Comment:
These lines seem to be mostly testing the same thing as the following test "does not make a network request if data is already in state" and can be removed.